src/components/nodes/utilities/shapeMetrics.js

```diff
@@ -0,0 +1,15 @@
+const shapeBottom = (shape) => shape.position().y + shape.size().height;
+const shapeLeft = (shape) => shape.position().x;
+const shapeCenterX = (shape) => shape.position().x + (shape.size().width / 2);
+const shapeCenterY = (shape) => shape.position().y + (shape.size().height / 2);
+const shapeRight = (shape) => shape.position().x + shape.size().width;
+const shapeTop = (shape) => shape.position().y;
+
+module.exports = {
```

Comment:
@chipit24 @lisa-fehr @tuantruong625 
Is it OK to do this here or do we standardise on individual export statements?

Reply:
We don't use `module.exports` anywhere else in our source code. We've standardized on individual import / exports statements, what's knows as "ES modules". The `module.exports` import counterpart is `require`, it's what's known as CommonJS modules.

It's unorthodox to mix the two so let's stick with individual named ES module exports.

Reply:
Looks like JointJS has methods for calculating all this on it's `Rect` object: https://resources.jointjs.com/docs/jointjs/v3.1/geometry.html#g.Rect.prototype. The only thing is, we'd have to create a rectangle first, like `const rect = g.Rect(x, y, width, height)`.

If it's possible to use that, we could get rid of all this code 😄

Reply:
That's fairly easy to do - I'm going to keep the basic signature the same so that we can get to a basic version of align & distribute and then look at stripping out the first-pass code.